src/overlays/effects/ovl_Effect_Ss_Fire_Tail/z_eff_ss_fire_tail.c

```diff
@@ -1,27 +1,142 @@
 /*
  * File: z_eff_ss_fire_tail.c
  * Overlay: ovl_Effect_Ss_Fire_Tail
- * Description:
+ * Description: Fire (burned by something)
  */
 
 #include "z_eff_ss_fire_tail.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+
+#define rScale regs[0]
+#define rLifespan regs[1]
+#define rReg2 regs[2]
+#define rReg3 regs[3]
+#define rPrimColorR regs[4]
+#define rPrimColorG regs[5]
+#define rPrimColorB regs[6]
+#define rEnvColorR regs[7]
+#define rEnvColorG regs[8]
+#define rEnvColorB regs[9]
+#define rReg10 regs[10]
+#define rBodyPart regs[11]
+#define rType regs[12]
 
 #define PARAMS ((EffectSsFireTailInitParams*)initParamsx)
 
-s32 EffectSsFireTail_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsFireTail_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsFireTail_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsFireTail_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_Fire_Tail_InitVars = {
     EFFECT_SS_FIRE_TAIL,
     EffectSsFireTail_Init,
 };
 
-#endif
+u32 EffectSsFireTail_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsFireTailInitParams* initParams = PARAMS;
+
+    this->pos = initParams->pos;
+    this->vec = initParams->unk_14;
+    this->velocity.x = 0.0f;
+    this->velocity.y = 0.0f;
+    this->velocity.z = 0.0f;
+    this->accel.x = 0.0f;
+    this->accel.y = 0.0f;
+    this->accel.z = 0.0f;
+    this->life = initParams->life;
+    this->actor = initParams->actor;
+    this->draw = EffectSsFireTail_Draw;
+    this->update = EffectSsFireTail_Update;
+    this->rScale = initParams->scale * 1000.0f;
+    this->rLifespan = initParams->life;
+    this->rReg2 = -10;
+    this->rReg3 = -15;
+    if (initParams->unk_20 == 0) {
+        initParams->unk_20 = 1;
+    }
+    this->rReg10 = initParams->unk_20;
+    this->rPrimColorR = initParams->primColor.r;
+    this->rPrimColorG = initParams->primColor.g;
+    this->rPrimColorB = initParams->primColor.b;
+    this->rEnvColorR = initParams->envColor.r;
+    this->rEnvColorG = initParams->envColor.g;
+    this->rEnvColorB = initParams->envColor.b;
+    this->rBodyPart = initParams->bodyPart;
+    this->rType = initParams->type;
+
+    return 1;
```

Comment:
er.. I forget is this bool?

Reply:
Probably, but just haven't started doing it yet